src/Piral.Blazor.Core/ComponentActivationService.cs

```diff
@@ -83,19 +91,55 @@ public void DeactivateComponent(string componentName)
         public void LoadComponentsFromAssembly(Assembly assembly)
         {
             var serviceProvider = _container.Configure(assembly);
-            var types = assembly.GetTypes().Where(m => m.GetCustomAttribute<ExposePiletAttribute>(false) != null);
-
-            foreach (var type in types)
+            var componentTypes = assembly.GetTypesWithAttributes(AttributeTypes);
+            
+            foreach (var componentType in componentTypes)
             {
-                var name = type.GetCustomAttribute<ExposePiletAttribute>(false).Name;
-                Register(name, type, serviceProvider);
+                var toRegister = GetComponentNamesToRegister(componentType, AttributeTypes);
+                foreach (string componentName in toRegister)
+                {
+                    if (componentName is null) continue;

```

Comment:
Can this be a real case? I think the `GetComponentNamesToRegister` should only return valid names.

Also maybe call it `componentNames` instead of `toRegister`. The latter is implied by the function.